docs/content/docs/reference/unittests.md

```diff
@@ -0,0 +1,160 @@
++++
+title = "Unit Tests"
+description = "Build unit tests for your modules"
+date = 2024-06-13T08:20:00+00:00
+updated = 2024-06-13T08:20:00+00:00
+draft = false
+weight = 110
+sort_by = "weight"
+template = "docs/page.html"
+
+[extra]
+toc = true
+top = false
++++
+
+## Create a context
+
+When writing a unit test, first create a context:
+```go
+func ExampleTest(t *testing.Test) {
+    ctx := ftltest.Context(
+        // options go here
+    )
+}
+```
+
+FTL will help isolate what you want to test by restricting access to FTL features by default. You can expand what is available to test by adding options to `ftltest.Context(...)`.
+
+In this default set up, FTL does the following:
+- prevents access to `ftl.ConfigValue` and `ftl.SecretValue` ([See options](#project-files-configs-and-secrets))
+- prevents access to `ftl.Database` ([See options](#databases))
+- prevents access to `ftl.MapHandle` ([See options](#maps))
+- prevents calls via `ftl.Call(...)` ([See options](#calls))
+- disables all subscribers ([See options](#pubsub))
+
+## Customization
+### Project files, configs and secrets
+
+To enable configs and secrets from the default project file:
+```go
+ctx := ftltest.Context(
+    ftltest.WithDefaultProjectFile(),
+)
+```
+
+Or you can specify a specific project file:
+```go
+ctx := ftltest.Context(
+    ftltest.WithProjectFile(path),
+)
+```
+
+You can also override specific config and secret values:
+```go
+ctx := ftltest.Context(
+    ftltest.WithDefaultProjectFile(),
+    
+    ftltest.WithConfig(endpoint, "test"),
+    ftltest.WithSecret(secret, "..."),
+)
+```
+
+### Databases
+By default, calling `Get(ctx)` on a database panics.
+
+To enable database access in a test, you must first [provide a DSN via a project file](#project-files-configs-and-secrets). You can then set up a test database:
+```go
+ctx := ftltest.Context(
+    ftltest.WithDefaultProjectFile(),
+    ftltest.WithDatabase(db),
+)
+```
+This will:
+- Take the provided DSN and appends `_test` to the database name. Eg: `accounts` becomes `accounts_test`
+- Wipe all tables in the database so each test run happens on a clean database
+
+
+### Maps
+By default, calling `Get(ctx)` on a map handle will panic.
+
+You can inject a fake via a map:
+```go
+ctx := ftltest.Context(
+    ftltest.WhenMap(exampleMap, func(ctx context.Context) (any, error) {
```

Comment:
```suggestion
    ftltest.WhenMap(exampleMap, func(ctx context.Context) (string, error) {
```

`WhenMap` is type safe.

Reply:
Almost was, but fixed here: https://github.com/TBD54566975/ftl/pull/1777
Will update here as well